addons/point_of_sale/static/src/app/screens/partner_list/partner_editor/partner_editor.js

```diff
@@ -48,13 +58,16 @@ export class PartnerDetailsEdit extends Component {
             return false;
         }
     }
-    /**
-     * Save to field `changes` all input changes from the form fields.
-     */
-    captureChange(event) {
-        this.changes[event.target.name] = event.target.value;
-    }
     saveChanges() {
+        this.changes.country_id = this.changes.country_id && parseInt(this.changes.country_id);
+        this.changes.state_id = this.changes.state_id && parseInt(this.changes.state_id);
+        if (
+            this.changes.state_id &&
+            this.pos.states.find((state) => state.id === this.changes.state_id).country_id[0] !==
+                this.changes.country_id
+        ) {
+            this.changes.state_id = false;
+        }
```

Comment:
Maybe instead of doing this here, we do it after populating `processedChanges` below. That way, `country_id`, `state_id` are already int.

Reply:
Good idea!